crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -869,6 +919,47 @@ impl<'a> TryFrom<&'a mut FilteredEntityMut<'_>> for EntityMut<'a> {
     }
 }
 
+impl PartialEq for EntityMut<'_> {
+    fn eq(&self, other: &Self) -> bool {
+        self.entity() == other.entity()
```

Comment:
I understand why we need to have this, but it's a little amusing since there can't possibly be two `EntityMut`s alive for the same `entity`.  It's also not very useful with `Query::iter_many`, since you can never have an `EntityMut` that would match the query.  Well, it can get combined with this other impls as part of the `EntityPtr` work and then it won't seem odd anymore.

Reply:
You still can have its entity match another query! That is one thing the `nesting_queries()` test in `EntitySet` shows. But it is definitely more obscure.

Reply:
Oh, I just meant `EntityMut` specifically.  It has write access to all components, so it's going to conflict with any other query that matches the same archetypes.  :)

Reply:
I also meant `EntityMut` specifically! You can still query for an archetype without accessing any components :)